Failed to build RPMs - gzip: unrecognized option '--keep'

When I'm trying to build anax RPMs on RHEL 7.9, it cannot finish a tgz file packaging after rpm build.
Here is the log obtained:
```
# export VERSION=2.27.0
# export BUILD_NUMBER=el7
# make rpmpkgs
Producing anax given arch: ppc64el
# github.com/open-horizon/anax
loadinternal: cannot find runtime/cgo
The required minimum exchange version is 2.44.0
The preferred exchange version is 2.44.0
Building target platforms: ppc64le-linux
Building for target ppc64le-linux
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.G57IZW
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd /root/rpmbuild/BUILD
+ rm -rf horizon-2.27.0
+ /usr/bin/mkdir -p horizon-2.27.0
+ cd horizon-2.27.0
+ /usr/bin/gzip -dc /root/rpmbuild/SOURCES/horizon-2.27.0.tar.gz
+ /usr/bin/tar -xvvf -
drwxr-xr-x root/root         0 2021-01-25 02:35 ./
-rw-r--r-- root/root       140 2021-01-25 02:35 ./.gitignore
-rw-r--r-- root/root      2494 2021-01-25 02:35 ./Makefile
drwxr-xr-x root/root         0 2021-01-26 08:50 ./fs/
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/etc/
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/etc/horizon/
-rw-r--r-- root/root       946 2021-01-25 02:35 ./fs/etc/horizon/anax.json
-rw-r--r-- root/root        58 2021-01-25 02:35 ./fs/etc/horizon/hzn.json
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/etc/horizon/trust/
-rw-r--r-- root/root      4910 2021-01-25 02:35 ./fs/etc/horizon/trust/horizon.pem
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/lib/
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/lib/systemd/
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/lib/systemd/system/
-rw-r--r-- root/root       396 2021-01-25 02:35 ./fs/lib/systemd/system/horizon.service
drwxr-xr-x root/root         0 2021-01-26 08:50 ./fs/usr/
drwxr-xr-x root/root         0 2021-01-26 08:50 ./fs/usr/horizon/
drwxr-xr-x root/root         0 2021-01-26 08:50 ./fs/usr/horizon/bin/
-rwxr-xr-x root/root  53288784 2021-01-26 09:55 ./fs/usr/horizon/bin/anax
drwxr-xr-x root/root         0 2021-01-25 02:35 ./fs/usr/horizon/samples/
-rw-r--r-- root/root      1037 2021-01-25 02:35 ./fs/usr/horizon/samples/business_policy.json
-rw-r--r-- root/root       962 2021-01-25 02:35 ./fs/usr/horizon/samples/insert-service-into-pattern.json
-rw-r--r-- root/root       397 2021-01-25 02:35 ./fs/usr/horizon/samples/node_policy_input.json
-rw-r--r-- root/root      1207 2021-01-25 02:35 ./fs/usr/horizon/samples/pattern.json
-rw-r--r-- root/root       811 2021-01-25 02:35 ./fs/usr/horizon/samples/service.json
-rw-r--r-- root/root       413 2021-01-25 02:35 ./fs/usr/horizon/samples/service_cluster.json
-rw-r--r-- root/root       497 2021-01-25 02:35 ./fs/usr/horizon/samples/user_input.json
-rw-r--r-- root/root     10141 2021-01-25 02:35 ./fs/usr/horizon/LICENSE.txt
-rw-r--r-- root/root      6130 2021-01-25 02:35 ./horizon.spec
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.mKvn5S
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd horizon-2.27.0
+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.X2UbbP
+ umask 022
+ cd /root/rpmbuild/BUILD
+ '[' /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le '!=' / ']'
+ rm -rf /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
++ dirname /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
+ mkdir -p /root/rpmbuild/BUILDROOT
+ mkdir /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
+ cd horizon-2.27.0
+ rm -rf /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
+ mkdir -p /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le/usr/horizon/bin /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le/usr/horizon/samples /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le/etc/default /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le/etc/horizon/trust/
+ cp -a fs/etc fs/lib fs/usr /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le/
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/redhat/brp-compress
+ /usr/lib/rpm/redhat/brp-strip /usr/bin/strip
+ /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip /usr/bin/objdump
+ /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python 1
+ /usr/lib/rpm/redhat/brp-python-hardlink
+ /usr/lib/rpm/redhat/brp-java-repack-jars
Processing files: horizon-2.27.0-el7.ppc64le
warning: File listed twice: /usr/horizon/LICENSE.txt
Provides: horizon = 1:2.27.0-el7 horizon = 2.27.0 horizon(ppc-64) = 1:2.27.0-el7
Requires(interp): /bin/sh /bin/sh /bin/sh
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires(post): /bin/sh
Requires(preun): /bin/sh
Requires(postun): /bin/sh
Requires: libc.so.6()(64bit) libc.so.6(GLIBC_2.17)(64bit) rtld(GNU_HASH)
Checking for unpackaged file(s): /usr/lib/rpm/check-files /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
Wrote: /root/rpmbuild/SRPMS/horizon-2.27.0-el7.src.rpm
Wrote: /root/rpmbuild/RPMS/ppc64le/horizon-2.27.0-el7.ppc64le.rpm
Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.IZgBMM
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd horizon-2.27.0
+ rm -rf /root/rpmbuild/BUILDROOT/horizon-2.27.0-el7.ppc64le
+ exit 0
gzip: unrecognized option '--keep'
Try `gzip --help' for more information.
make[2]: *** [rpmbuild] Error 1
make[1]: *** [all] Error 2
make: *** [rpmpkgs] Error 2
```

Version of GNU gzip tool on that host:
```
# gzip -V
gzip 1.5
Copyright (C) 2007, 2010, 2011 Free Software Foundation, Inc.
Copyright (C) 1993 Jean-loup Gailly.
This is free software.  You may redistribute copies of it under the terms of
the GNU General Public License <http://www.gnu.org/licenses/gpl.html>.
There is NO WARRANTY, to the extent permitted by law.

Written by Jean-loup Gailly.
```

--- a/pkg/rpm/horizon-cli/Makefile
+++ b/pkg/rpm/horizon-cli/Makefile
@@ -71,7 +71,12 @@ rpmbuild: require-version
 	rm -rf $(RPMROOT)/BUILD/horizon-cli-*
 	rm -f $(RPMROOT)/SRPMS/$(RPMNAME)*rpm $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)*rpm $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)*rpm.gz
 	rpmbuild --target $(rpm_arch)-linux $(RPM_OPTS) -ba $(RPMNAME).spec
-	gzip --keep $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$(VERSION)-$(BUILD_NUMBER).$(rpm_arch).rpm
+	$(eval target_rpm := $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$(VERSION)-$(BUILD_NUMBER).$(rpm_arch).rpm)
+	@# There is a '--keep' flag available in gzip >= 1.6 for keeping original file (.rpm), but to
+	@# save compatibility with previous versions of gzip tool (older than 1.6, e.g.
+	@# RHEL 7.x has 1.5 basically), it's recommended to use simple shell redirects
+	@# See https://unix.stackexchange.com/a/58814
+	gzip < $(target_rpm) > $(target_rpm).gz
 	rm -f $(RPMNAME)-$(VERSION)   # remove the sym link
 
 .PHONY: rpmbuild

--- a/pkg/rpm/horizon/Makefile
+++ b/pkg/rpm/horizon/Makefile
@@ -58,7 +58,12 @@ rpmbuild: require-version
 	rm -rf $(RPMROOT)/BUILD/horizon-*
 	rm -f $(RPMROOT)/SRPMS/$(RPMNAME)-$${VERSION%%.*}*rpm $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$${VERSION%%.*}*rpm $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$${VERSION%%.*}*rpm.gz
 	rpmbuild --target $(rpm_arch)-linux $(RPM_OPTS) -ba $(RPMNAME).spec
-	gzip --keep $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$(VERSION)-$(BUILD_NUMBER).$(rpm_arch).rpm
+	$(eval target_rpm := $(RPMROOT)/RPMS/$(rpm_arch)/$(RPMNAME)-$(VERSION)-$(BUILD_NUMBER).$(rpm_arch).rpm)
+	@# There is a '--keep' flag available in gzip >= 1.6 for keeping original file (.rpm), but to
+	@# save compatibility with previous versions of gzip tool (older than 1.6, e.g.
+	@# RHEL 7.x has 1.5 basically), it's recommended to use simple shell redirects
+	@# See https://unix.stackexchange.com/a/58814
+	gzip < $(target_rpm) > $(target_rpm).gz
 	rm -f $(RPMNAME)-$(VERSION)   # remove the sym link
 
 .PHONY: rpmbuild

